src/badguy/badguy.cpp

```diff
@@ -170,11 +183,24 @@ BadGuy::draw(DrawingContext& context)
       }
       else
       {
-        if (m_frozen && is_portable())
+        if (m_frozen && is_portable()) {
           m_freezesprite->draw(context.color(), draw_pos, m_layer);
+        }
+
+        if (m_state != STATE_BURNING || m_firesprite->get_current_frame() < 5)
         m_sprite->draw(context.color(), draw_pos, m_layer - (m_frozen ? 1 : 0), m_flip);
       }
 
+      if (m_state == STATE_BURNING) {
+        // draw the flame sprite
+        m_firesprite->draw(context.color(), draw_pos, m_layer);
+        m_firesprite->set_action(get_overlay_size(), 1);
```

Comment:
Is it possible to do this outside of the draw function? set_action compares strings and does stuff that isn't adequate in a draw function.